modules/concepts/controllers/admin-controllers/_index.md

```diff
@@ -9,75 +9,179 @@ Using modern pages, you will have access to the PrestaShop debug toolbar, the se
+
+### Option 1: Explicit service configuration with tags
 
-If you want Symfony Dependency Injection to inject services into your controller, you need to use specific YAML service declaration:
 ```yaml
+# modules/your-module/config/services.yml
 services:
-  # The name of the service must match the full namespace class
   MyModule\Controller\DemoController:
-    class: MyModule\Controller\DemoController
-    arguments:
-      - '@doctrine.cache.provider'
+    autowire: true
+    autoconfigure: true
+    tags:
+      - { name: controller.service_arguments }
 ```
 
-You can also retrieve services with the container available in symfony controllers ->
-```php
-<?php
-// modules/your-module/src/Controller/DemoController.php
+### Option 2: Using service defaults
 
-namespace MyModule\Controller;
+```yaml
+# modules/your-module/config/services.yml
+services:
+  _defaults:
+    public: false
+    autowire: true
+    autoconfigure: true
+  
+  MyModule\Controller\DemoController: ~
+```
 
-use Doctrine\Common\Cache\CacheProvider;
-use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+{{% notice warning %}}
+**Service configuration is mandatory:**
 
-class DemoController extends FrameworkBundleAdminController
-{
-    public function demoAction()
-    {
-        // you can also retrieve services directly from the container
-        $cache = $this->container->get('doctrine.cache');
-        
-        return $this->render('@Modules/your-module/templates/admin/demo.html.twig');
-    }
-}
-```
+One of the two service configuration options above is **essential and required** for your controller to work properly. Without this configuration:
+- **The controller will not function** and will throw errors
+- Helper methods from `PrestaShopAdminController` will not be accessible
+- Services like Twig, Form, Router, and other Symfony components will not be available
 
-You have access to the Container, to Twig as rendering engine, the Doctrine ORM, everything from Symfony framework ecosystem.
-Note that you must return a `Response` object, but this can be a `JsonResponse` if you plan to make a single page application (or "SPA").
+**Important:** The service name (e.g., `MyModule\Controller\DemoController`) **must exactly match** the fully qualified class name (FQCN) of your controller. If the service name doesn't match the class name, Symfony and PrestaShop will not be able to identify and instantiate the controller, resulting in errors.
 
-{{% notice note %}}
-This controller works exactly the same as the Core Back Office ones.
-{{% /notice %}} 
+**Understanding the configuration:**
+- `autowire: true` - Automatically injects services in constructors and method parameters
+- `autoconfigure: true` - Automatically configures the controller as a service and enables all controller features
+- `controller.service_arguments` tag - Required if your controller doesn't extend `AbstractController` to enable method parameter injection
```

Comment:
It's true! But should we mention this? And by that I mean should we encourage people having controller that do not extend `AbstractController`, so not `PrestaShopAdminController` in the end

Or should we simply document: Use this class `PrestaShopAdminController` and that's enough documentation because any other type of controllers means you know what you're doing with the Symfony framework, so you are on your own because it's your specific need and your expertise